controllers/gateway/apirule_controller.go

```diff
@@ -295,9 +300,25 @@ func (r *APIRuleReconciler) updateStatus(ctx context.Context, api *gatewayv1beta
 	api.Status.AuthorizationPolicyStatus = status.AuthorizationPolicyStatus
 
 	r.Log.Info("Updating ApiRule status", "status", api.Status)
-	err := r.Client.Status().Update(ctx, api)
+	err = r.Client.Status().Update(ctx, api)
```

Comment:
can we have a retry in here, like in Istio:
https://github.com/kyma-project/istio/blob/88aa69cd79b7ca3cf788e8a038ca8bc851809eef/controllers/status.go#L28-L45

we could retry setting the status, not needed to requeue whole reconcile for it when it fails:
https://github.com/kyma-project/api-gateway/blob/131495f2086a9b7a0ff896cb5f5768824ea1989f/controllers/gateway/apirule_controller.go#L246